src/Tools/ExternalAccess/Razor/Features/Cohost/RazorStartupServiceFactory.cs

```diff
@@ -82,10 +89,18 @@ private async Task InitializeRazorAsync(ClientCapabilities clientCapabilities, R
 
             // We use a string to pass capabilities to/from Razor to avoid version issues with the Protocol DLL
             var serializedClientCapabilities = JsonSerializer.Serialize(clientCapabilities, ProtocolConversions.LspJsonSerializerOptions);
-            var razorCohostClientLanguageServerManager = new RazorClientLanguageServerManager(clientLanguageServerManager!);
 
             var requestContext = new RazorCohostRequestContext(context);
-            await dynamicRegistrationService!.Value.RegisterAsync(serializedClientCapabilities, requestContext, cancellationToken).ConfigureAwait(false);
+
+            if (cohostStartupService is not null)
+            {
+                await cohostStartupService.Value.StartupAsync(serializedClientCapabilities, requestContext, cancellationToken).ConfigureAwait(false);
```

Comment:
consider trace logging on which service actually gets used, in case we need to debug something

Reply:
The two services are just dual-insertion prevention. I'm going to remove the old one as soon as https://github.com/dotnet/razor/pull/11748 merges.